Support standalone dvc installation via metapackages.

asford opened the pull request:
_closes #160_
_adds @asford to maintainers to support metapackage layout_ 

Build a `dvc` conda package equivalent to the `dvc[all]` pip
package, with a `_dvc` base package containing dvc alone. This allows a
happy-path of:

  `conda install -c conda-forge dvc`

supporting dvc and all remotes and a power-user path of:

  `conda install -c conda-forge _dvc _dvc-s3`

for dvc and specific remotes.

See comments in meta.yml for details.

-----------------------

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

efiop commented:
Thank you @asford ! Looks neat! :sparkles: 

Btw, looks like windows is failing, could you take a look, please?

asford commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * The recipe must have some tests.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

asford commented:
@conda-forge-admin, please rerender

asford commented:
@conda-forge-admin, please rerender

asford commented:
@efiop I've gone ahead and fixed the windows build and polished this up for the smithy and linter. Would you mind reviewing the `meta.yaml` to ensure that you understand the core logic?

The interaction between the smithy/linter/build was a *little* persnickety, as there are multiple equivalent ways of phrasing this for conda-build but only one or two that dual-satisfy the linter and smithy.  I've added myself to the maintainers list so I can proactively reach out with any issues due to the metapackage layout. Feel free to revert bfb745229b74e4715fbada221e2e394aebfb4adb if you'd prefer to keep the maintainers list short.

efiop commented on recipe/meta.yaml:
```diff
@@ -11,70 +11,179 @@ source:
 build:
-  number: 0
-  script: {{ PYTHON }} -m pip install . -vv
-  skip: true  # [py<36]
```
Did we lose this?

asford replied:
I *think* conda-forge doesn't target `py<36` anymore, so we weren't seeing a delta in the build matrix, but I've reintroduced for the sake of clarity in the recipe.

efiop commented on recipe/meta.yaml:
```diff
@@ -11,70 +11,179 @@ source:
+# We define extras through "private" subpackages and a single top-level `dvc`
+# metapackage. The metapackage at a given build version references the extras
+# subpackages at the same build version. `_dvc` includes the dvc package itself
+# and core dependencies. `_dvc-gs`, `_dv-s3`, et al contain no code, but
```
```suggestion
# and core dependencies. `_dvc-gs`, `_dvc-s3`, et al contain no code, but
```

efiop commented:
@asford Thank you so much! Looks great! Just a few minor comments above.

asford commented on recipe/meta.yaml:
```diff
@@ -11,70 +11,179 @@ source:
+# Of the build properties here and below:
+# noarch is not inherited and must be specified per output
+# number is propagated into the outputs below
 build:
```
```suggestion
build:
  skip: true # [py<36]
```

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [18]

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

asford commented:
@conda-forge-admin, please rerender

asford commented:
@efiop Thanks for the catch on the review. This should be good-to-merge pending final build passes.

efiop commented:
Thank you @asford ! :pray:
